openedx/features/badging/models.py

```diff
@@ -145,3 +145,28 @@ def assign_badge(cls, user_id, badge_id, community_id):
             error = badge_constants.BADGE_TYPE_ERROR.format(badge_id=badge_id, badge_type=badge_type)
             log.exception(error)
             raise Exception(error)
+
+    @classmethod
+    def assign_missing_team_badges(cls, user_id, team_id):
+        """
+        Assign all previous (missing) badges when user joins a team,
+        such that he has same number of badges as any other member
+        of same team
+        """
+        if user_id is None or team_id is None:
+            error = badge_constants.TEAM_BADGE_ERROR.format(user_id=user_id, team_id=team_id)
+            log.exception(error)
+            return Exception(error)
+
+        team_group_chat = TeamGroupChat.objects.filter(team_id=team_id).values(badge_constants.ROOM_ID_KEY).first()
+        earned_team_badges = UserBadge.objects.filter(community_id=team_group_chat[badge_constants.ROOM_ID_KEY]).values(
```

Comment:
this query is a little complex can we have a comment to explain it and better spacing?